sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/schema/BeamTableUtils.java

```diff
@@ -149,13 +149,35 @@ public static Object autoCastField(Schema.Field field, Object rawObj) {
         case INT16:
           return Short.valueOf(raw);
         case INT32:
+          if (raw.equals("")) {
```

Comment:
seems to me that this is not the place where you should handle "" to null conversion.

Reply:
Removed modifications on case INT32, FLOAT and DECIMAL.